### Задачи 1 и 2

#### Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Случайная величина X задаёт число правильных ответов при простом угадывании. Найдите математическое ожидание данной случайной величины. Найдите дисперсию случайной величины X.

#### Подсказка: постройте закон распределения случайной величины X.

In [1]:
import numpy as np

Пусть Х - величина, равная количеству правильных ответов. При этом вероятность угадать один правильный ответ из 4-х будет равна: $$p=\frac{1}{4}$$

n - количество вопросов. В нашем случае n = 5

Используем формулу Бернули для определения вероятности наступления каждого из событий (ни один ответ не угадан, угадан один ответ, угадано два ответа и т.д)

$$P(X = k) = C_n^k p^k (1 - p)^{n - k}$$

$$P(X=0)=C_5^0 \cdot (1/4)^0 \cdot (3/4)^{5-0} $$
$$P(X=1)=C_5^1 \cdot (1/4)^1 \cdot (3/4)^{5-1} $$
$$P(X=2)=C_5^2 \cdot (1/4)^2 \cdot (3/4)^{5-2} $$
$$P(X=3)=C_5^3 \cdot (1/4)^3 \cdot (3/4)^{5-3} $$
$$P(X=4)=C_5^4 \cdot (1/4)^4 \cdot (3/4)^{5-4} $$
$$P(X=5)=C_5^5 \cdot (1/4)^5 \cdot (3/4)^{5-5} $$

Подставляем значения и считаем вероятности событий

In [2]:
p_x0 = (np.math.factorial(5) / (np.math.factorial(0) * np.math.factorial(5 - 0))) * 0.25 ** 0 * 0.75 ** 5
p_x1 = (np.math.factorial(5) / (np.math.factorial(1) * np.math.factorial(5 - 1))) * 0.25 ** 1 * 0.75 ** 4
p_x2 = (np.math.factorial(5) / (np.math.factorial(2) * np.math.factorial(5 - 2))) * 0.25 ** 2 * 0.75 ** 3
p_x3 = (np.math.factorial(5) / (np.math.factorial(3) * np.math.factorial(5 - 3))) * 0.25 ** 3 * 0.75 ** 2
p_x4 = (np.math.factorial(5) / (np.math.factorial(4) * np.math.factorial(5 - 4))) * 0.25 ** 4 * 0.75 ** 1
p_x5 = (np.math.factorial(5) / (np.math.factorial(5) * np.math.factorial(5 - 5))) * 0.25 ** 5 * 0.75 ** 0

In [3]:
print(p_x0)
print(p_x1)
print(p_x2)
print(p_x3)
print(p_x4)
print(p_x5)

0.2373046875
0.3955078125
0.263671875
0.087890625
0.0146484375
0.0009765625


Проверяем чтобы сумма значений вероятностей равнялась единице:

In [4]:
p_x0 + p_x1 + p_x2 + p_x3 + p_x4 + p_x5

1.0

Получили закон распределения

<table border="3">
    <tr>
        <th>$x_i$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
</tr>
    <tr>
        <th>$P_i$</th>
        <td>0.2373</td>
        <td>0.3955</td>
        <td>0.2636</td>
        <td>0.0878</td>
        <td>0.0146</td>
        <td>0.0009</td>
    </tr>
</table>

##### Определяем математическое ожидание

Т.к. вероятность выбора правильного ответа в каждом вопросе у нас одинакова и равна $ \frac{1}{4} $, находим математическое ожидание биноминального распределения

$$ M(X) = n \cdot p = 5 \cdot \frac{1}{4} = \frac{5}{4} = 1,25 $$

#### Определяем дисперсию случайной величины Х

$$ D(X) = n \cdot p \cdot (1 - p) = 5 \cdot \frac{1}{4} \cdot \frac{3}{4} = \frac{15}{16} = 0,9375 $$

### Задача 3

#### Пользователь получает в среднем 10 писем со спамом на свой почтовый ящик за сутки. Найти число N, такое, что с вероятностью 0.95 пользователь получит не более N писем со спамом за текущий день.

Для решения задачи воспользуемся формулой Пуасона: $$P(X = k) = \dfrac{\lambda^k e^{-\lambda}}{k!}$$

где $ \lambda = 10 $

Вероятность получить не более N писем предполагает получение от 0 до N писем. При этом нужно будет cложить несколько вероятностей $$P(X \leq N) = \displaystyle\sum_{i = 0}^{N} \dfrac{10^{i} e^{-10}}{i!}$$ При сумме вероятностей, равной 0,95 мы узнаем чему равно N писем

In [5]:
def poisson_proba(k: int, lambda_: float) -> float:
    """Формула Пуассона.
    """
    
    return (lambda_ ** k) * (np.exp(-lambda_)) / np.math.factorial(k)

In [6]:
lambda_ = 10

In [7]:
my_dict = []
for i in range(0, 20):
    a = poisson_proba(k=i, lambda_=lambda_)
    my_dict.append(a)
    print(i, round((sum(my_dict)),4))
    i += 1

0 0.0
1 0.0005
2 0.0028
3 0.0103
4 0.0293
5 0.0671
6 0.1301
7 0.2202
8 0.3328
9 0.4579
10 0.583
11 0.6968
12 0.7916
13 0.8645
14 0.9165
15 0.9513
16 0.973
17 0.9857
18 0.9928
19 0.9965


#### Ответ: пользователь получит не более 15 писем

### Задача 4

#### Производятся выстрелы по мишени. Вероятность попадания в мишень при одном выстреле равна 0.01. Сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 6 раз будет совершено попадание?

#### Подсказка: 1) "Вероятность попасть k раз при n выстрелах" - на какое распределение это похоже? 2) А если нам нужна вероятность P(X >= k), а не P(X = k)? 3) Здесь предстоит немножко покодить.

Исходя из условия задачи, каждый выстрел рассматривается как независимое событие. Для решения будем использовать формулу Бернули $$P(X = k) = C_n^k p^k (1 - p)^{n - k}$$

Для решения задачи будем использовать формулу для вероятности противоположного события $$P_n(X>=6) = 1 - P_n(X<6)$$

где $P_n(X<6)$ - суммарная вероятность попаданий от 0 до 5 раз. Обозначим $P_n$

Введем функцию для расчёта количества сочетаний

In [8]:
def combinations(n: int, k: int) -> int:
        
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

In [9]:
p = 0.01
k = 6
n = 6
P = 0

while (P < 0.9):
    k = 0 # Расчет вероятности для 0 попаданий
    p_0 = combinations(n, k) * (p ** k) * ((1 - p)**(n - k))
    k = 1 # Расчет вероятности для 1 попадания
    p_1 = combinations(n, k) * (p ** k) * ((1 - p)**(n - k))
    k = 2 # Расчет вероятности для 2 попаданий
    p_2 = combinations(n, k) * (p ** k) * ((1 - p)**(n - k))
    k = 3 # Расчет вероятности для 3 попаданий
    p_3 = combinations(n, k) * (p ** k) * ((1 - p)**(n - k))
    k = 4 # Расчет вероятности для 4 попаданий
    p_4 = combinations(n, k) * (p ** k) * ((1 - p)**(n - k))
    k = 5 # Расчет вероятности для 5 попаданий
    p_5 = combinations(n, k) * (p ** k) * ((1 - p)**(n - k))
    
    P_n = p_0 + p_1 + p_2 + p_3 + p_4 + p_5
    P = (1 - P_n)
    print(f'номер_выстрела: {n}, P(X>=6):{P}')
    n +=1

номер_выстрела: 6, P(X>=6):9.999778782798785e-13
номер_выстрела: 7, P(X>=6):6.940115149234316e-12
номер_выстрела: 8, P(X>=6):2.752198469124778e-11
номер_выстрела: 9, P(X>=6):8.185885302935958e-11
номер_выстрела: 10, P(X>=6):2.028940349063646e-10
номер_выстрела: 11, P(X>=6):4.42543446332877e-10
номер_выстрела: 12, P(X>=6):8.775072890543356e-10
номер_выстрела: 13, P(X>=6):1.6157031268448918e-09
номер_выстрела: 14, P(X>=6):2.8032755006179855e-09
номер_выстрела: 15, P(X>=6):4.632137118321111e-09
номер_выстрела: 16, P(X>=6):7.347996544559976e-09
номер_выстрела: 17, P(X>=6):1.1258834087257696e-08
номер_выстрела: 18, P(X>=6):1.67437835818518e-08
номер_выстрела: 19, P(X>=6):2.4262383657713826e-08
номер_выстрела: 20, P(X>=6):3.436415985103025e-08
номер_выстрела: 21, P(X>=6):4.769850470154324e-08
номер_выстрела: 22, P(X>=6):6.502481897374679e-08
номер_выстрела: 23, P(X>=6):8.722288513673959e-08
номер_выстрела: 24, P(X>=6):1.1530343868582094e-07
номер_выстрела: 25, P(X>=6):1.5041890988332085e-07


номер_выстрела: 622, P(X>=6):0.5899264391778044
номер_выстрела: 623, P(X>=6):0.5914741385952171
номер_выстрела: 624, P(X>=6):0.593018757640003
номер_выстрела: 625, P(X>=6):0.5945602824560078
номер_выстрела: 626, P(X>=6):0.596098699359077
номер_выстрела: 627, P(X>=6):0.5976339948365459
номер_выстрела: 628, P(X>=6):0.5991661555467215
номер_выстрела: 629, P(X>=6):0.6006951683183593
номер_выстрела: 630, P(X>=6):0.6022210201501327
номер_выстрела: 631, P(X>=6):0.603743698210096
номер_выстрела: 632, P(X>=6):0.6052631898351415
номер_выстрела: 633, P(X>=6):0.6067794825304501
номер_выстрела: 634, P(X>=6):0.6082925639689358
номер_выстрела: 635, P(X>=6):0.6098024219906845
номер_выстрела: 636, P(X>=6):0.6113090446023866
номер_выстрела: 637, P(X>=6):0.612812419976764
номер_выстрела: 638, P(X>=6):0.6143125364519911
номер_выстрела: 639, P(X>=6):0.6158093825311112
номер_выстрела: 640, P(X>=6):0.6173029468814457
номер_выстрела: 641, P(X>=6):0.6187932183340001
номер_выстрела: 642, P(X>=6):0.6202801858828

номер_выстрела: 850, P(X>=6):0.851723596249975
номер_выстрела: 851, P(X>=6):0.8524726869933563
номер_выстрела: 852, P(X>=6):0.8532186698070576
номер_выстрела: 853, P(X>=6):0.8539615524324422
номер_выстрела: 854, P(X>=6):0.8547013426336906
номер_выстрела: 855, P(X>=6):0.8554380481973506
номер_выстрела: 856, P(X>=6):0.8561716769318919
номер_выстрела: 857, P(X>=6):0.8569022366672616
номер_выстрела: 858, P(X>=6):0.8576297352544444
номер_выстрела: 859, P(X>=6):0.8583541805650244
номер_выстрела: 860, P(X>=6):0.8590755804907508
номер_выстрела: 861, P(X>=6):0.8597939429431058
номер_выстрела: 862, P(X>=6):0.8605092758528756
номер_выстрела: 863, P(X>=6):0.8612215871697242
номер_выстрела: 864, P(X>=6):0.8619308848617708
номер_выстрела: 865, P(X>=6):0.8626371769151688
номер_выстрела: 866, P(X>=6):0.8633404713336892
номер_выстрела: 867, P(X>=6):0.8640407761383049
номер_выстрела: 868, P(X>=6):0.8647380993667804
номер_выстрела: 869, P(X>=6):0.8654324490732619
номер_выстрела: 870, P(X>=6):0.8661238333

#### Ответ: 926 выстрелов